dags/leanplum_export.py

```diff
@@ -17,47 +17,62 @@
     "retry_delay": timedelta(minutes=30),
 }
 
+# tuples of (app_name, s3_prefix)
+LEANPLUM_APPS = [
+    (
+        "firefox_android_release",
+        "firefox_android",
+        "{{ var.value.LEANPLUM_FIREFOX_ANDROID_RELEASE_APP_ID }}",
+        "{{ var.value.LEANPLUM_FIREFOX_ANDROID_RELEASE_CLIENT_KEY }}"
+    ),
+    (
+        "firefox_android_beta",
+        "firefox_android_beta",
+        "{{ var.value.LEANPLUM_FIREFOX_ANDROID_BETA_APP_ID }}",
+        "{{ var.value.LEANPLUM_FIREFOX_ANDROID_BETA_CLIENT_KEY }}"
+    ),
+    (
+        "firefox_android_nightly",
+        "firefox_android_nightly",
+        "{{ var.value.LEANPLUM_FIREFOX_ANDROID_NIGHTLY_APP_ID }}",
+        "{{ var.value.LEANPLUM_FIREFOX_ANDROID_NIGHTLY_CLIENT_KEY }}"
+    ),
+    (
+        "firefox_ios_release",
+        "firefox_ios",
+        "{{ var.value.LEANPLUM_FIREFOX_IOS_RELEASE_APP_ID }}",
+        "{{ var.value.LEANPLUM_FIREFOX_IOS_RELEASE_CLIENT_KEY }}"
+    ),
+]
 
-with DAG("leanplum_export",
-         default_args=default_args,
-         schedule_interval=timedelta(hours=4)) as dag:
+with DAG(
+        "leanplum_export",
+        default_args=default_args,
+        # last run of a day will be at 03:59 of the next day
+        schedule_interval="59 3/4 * * *",
+        max_active_runs=1,
```

Comment:
Why limit active runs?

Reply:
Oh right, this has to do with the exports and not the messages.  It's because the next run may start before the previous one ends if there are a bunch of sequential airflow `Pod took to long to start` errors or for some other reason.  If that happens the second run will try to process files that the first job has in queue because a file isn't marked as done until it's processed.  This makes it so each file is only processed once